internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -548,7 +548,7 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 
 	updated := false
 	if string(conf) != infraEnvInternal.InternalIgnitionConfigOverride {
-		_, err = r.Installer.UpdateInfraEnvInternal(ctx, installer.UpdateInfraEnvParams{InfraEnvID: *infraEnvInternal.ID, InfraEnvUpdateParams: &models.InfraEnvUpdateParams{}}, swag.String(string(conf)))
+		_, err = r.Installer.UpdateInfraEnvInternal(ctx, installer.UpdateInfraEnvParams{InfraEnvID: *infraEnvInternal.ID, InfraEnvUpdateParams: &models.InfraEnvUpdateParams{}}, swag.String(string(conf)), nil)
```

Comment:
In a previous [comment](https://github.com/openshift/assisted-service/pull/6868#discussion_r1819651450), you mentioned that it is safe to remove this logic from here (or did I misunderstand your comment?).

Reply:
That was before we moved the mirror registries to the database.

Ideally it still true, we shouldn't have to care about setting the mirror registries when we're just trying to update the internal override.

I'm not sure how to handle this while also allowing the user to clear these registries. Maybe the internal override and/or the mirror registries should be set with a function separate from this update one?

Alternatively maybe we need some kind of test to make sure `UpdateInfraEnvInternal` doesn't clear the mirror registries when a nil is passed like this if that's the intention (because it sure looks like it will to me).